src/borg/archiver.py

```diff
@@ -1024,7 +1024,8 @@ def item_to_tarinfo(item, original_path):
                 return None, stream
             return tarinfo, stream
 
-        for item in archive.iter_items(filter, preload=True, hardlink_masters=hardlink_masters):
+        for item in archive.iter_items(filter, partial_extract=partial_extract,
+                                       preload=True, hardlink_masters=hardlink_masters):
```

Comment:
From above:
```
partial_extract = not matcher.empty() or strip_components
```

So this fix is only relevant for cases when this gets True.

Like: giving any options that lead to partial extraction of an archive, including/excluding stuff, stripping directory layers.

Reply:
Good catch.

Reply:
yeah, but the main fix is in last commit.